ruoyi-common/src/test/java/com/ruoyi/common/support/ConvertTest.java

```diff
@@ -0,0 +1,205 @@
+import static org.junit.Assert.*;
+
+public class ConvertTest {
+
+    @Test
+    public void testToStr() {
+        assertEquals("bar", toStr(null, "bar"));
+        assertEquals("123", toStr(123, "bar"));
+        assertEquals("foo", toStr("foo"));
+    }
+
+    @Test
+    public void testToChar() {
+        assertEquals('b', (char) toChar(null, 'b'));
+        assertEquals('1', (char) toChar(1, 'b'));
+        assertEquals('f', (char) toChar('f'));
+    }
+
+    @Test
+    public void testToByte() {
+        assertEquals((byte) 8, (byte) toByte(null, (byte) 8));
+        assertEquals((byte) 1, (byte) toByte((byte) 1, (byte) 8));
+        assertEquals((byte) 8, (byte) toByte("", (byte) 8));
+        assertEquals((byte) 123, (byte) toByte("123", (byte) 8));
+        assertEquals((byte) 8, (byte) toByte("foo", (byte) 8));
+        assertEquals((byte) 1, (byte) toByte(1));
+    }
+
+    @Test
+    public void testToShort() {
+        assertEquals((short) 8, (short) toShort(null, (short) 8));
+        assertEquals((short) 1, (short) toShort((short) 1, (short) 8));
+        assertEquals((short) 8, (short) toShort("", (short) 8));
+        assertEquals((short) 123, (short) toShort("123", (short) 8));
+        assertEquals((short) 8, (short) toShort("foo", (short) 8));
+        assertEquals((short) 1, (short) toShort(1));
+    }
+
+    @Test
+    public void testToNumber() {
+        assertEquals(8, toNumber(null, 8));
+        assertEquals(8, toNumber("", 8));
+        assertEquals((long) 123, toNumber("123", 8));
+        assertEquals(8, toNumber("foo", 8));
+        assertEquals(1, toNumber(1));
+    }
+
+    @Test
+    public void testToInt() throws ParseException {
+        assertEquals(8, (int) toInt(null, 8));
+        assertEquals(1, (int) toInt(NumberFormat.getInstance().parse("1"), 8));
+        assertEquals(8, (int) toInt("", 8));
+        assertEquals(123, (int) toInt("123", 8));
+        assertEquals(8, (int) toInt("foo", 8));
+        assertEquals(1, (int) toInt(1));
+    }
+
+    @Test
+    public void testToIntArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toIntArray(null, "")));
+        assertEquals(Arrays.asList(0, 0, 0, 0), Arrays.asList(toIntArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToLongArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toLongArray(null, "")));
+        assertEquals(Arrays.asList(null, null, null, null), Arrays.asList(toLongArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToStrArray() {
+        assertEquals(Arrays.asList("a", "b", "c", "d"), Arrays.asList(toStrArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToLong() {
+        assertEquals(8, (long) toLong(null, (long) 8));
+        assertEquals(1, (long) toLong((long) 1, (long) 8));
+        assertEquals(8, (long) toLong("", (long) 8));
+        assertEquals(123, (long) toLong("123", (long) 8));
+        assertEquals(8, (long) toLong("foo", (long) 8));
```

Comment:
Instead of casting to `(long)` in the second param of `toLong`, you could use a long literal (`8L`)

Reply:
Done